src/Common/MSBuildCachePluginBase.cs

```diff
@@ -49,6 +49,9 @@ public abstract class MSBuildCachePluginBase<TPluginSettings> : ProjectCachePlug
     private string? _repoRoot;
     private string? _buildId;
 
+    // Set if we've received any file access report. Ideally MSBuild would tell us in the CacheContext
+    private bool _hasHadFileAccessReport;
```

Comment:
Should this be inside `FileAccessRepository` ?

Reply:
So `MSBuildCachePluginBase` gets the call from MSBuild directly and forwards to `FileAccessRepository`, who would then expose a bool which `MSBuildCachePluginBase` would read to understand if it a file access report ever happened (which is would have forwarded =P). Seems a bit roundabout.

Note that this isn't checking per-project if a file access was ever reported, in which case `FileAccessRepository` would be a good place to track that, but instead whether any report has happened during the entire lifetime of the build.